addons/base_vat/models/res_partner.py

```diff
@@ -221,6 +221,9 @@ def _run_vat_test(self, vat_number, default_country, partner_is_company=True):
 
         # If it fails, check with default_country (if it exists)
         if default_country:
+            if not default_country.code:
+                msg = self._build_vat_error_message(vat_country_code, vat_number, self.name)
```

Comment:
I am not convinced by this solution. Reading the error message:
> The  number [BE123] for SuperContact does not seem to be valid. 
Note: the expected format is BE0477472701

The user would believe that he did a mistake in the VAT number, and that's not true (if he then tries with _BE0477472701_, he will get the same error)

Reply:
Changes are done please check.